src/PostgREST/Main.hs

```diff
@@ -71,6 +67,10 @@ main = do
         fail "Cannot run in this PostgreSQL version, PostgREST needs at least 9.2.0"
     ) supportedOrError
 
+  Right authenticator <- H.session pool $ do
```

Comment:
I know the select session user is right, I'm just a Haskell noob and I'm not sure how to handle the `Left` error and extract the `Right` value…